src/reformation/multitable.cljc

```diff
@@ -11,75 +11,103 @@
-                                             :disabled (or (:disabled c) disabled)
-                                             :placeholder (c :placeholder)
-                                             :default-value (c :default-value)
-                                             :on-change on-change-function
-                                             :name nameval
-                                             :value (READ multi-vpath)}]
-                                           (if-checkbox [:span.custom-control-indicator])]])))))
-        tbody (if-not sum-field tbody-base
-                      (let [sum-key (-> sum-field name (str "-total") keyword)
-                            sum-val (READ [sum-key])
-                            total (apply + (for [c (READ vpath)]
-                                             #?(:cljs (if-let [c (sum-field c)]
-                                                        (js/parseInt c)
-                                                        0)
-                                                :clj (if-let [c (sum-field c)]
-                                                       (Integer/parseInt c)
-                                                       0))))]
-                        (UPDATE [sum-key] (constantly total))
-                        (conj tbody-base
-                              [:tr [:td.form-group.total {:col-span 2}
-                                    [:label "Total"]
-                                    [:div.input-group
-                                     [:span.input-group-addon "$"]
-                                     [:input
-                                      {:type "text"
-                                       :disabled true
-                                       :value sum-val}]]]])))
-        add-button [:a.btn.btn-success
-                    {:on-click #(UPDATE vpath add-row row-template)}
-                    [:i.fa.fa-plus]]
-        delete-button [:a.btn.btn-danger
-                       {:on-click #(UPDATE vpath delete-row min-rows)}
-                       [:i.fa.fa-minus]]]
+
+        universals {:vpath vpath :fn-map fn-map :disabled disabled}
+        headers [:thead [:tr
+                         [:<> (for [{:keys [disabled title]} columns]
+                                [:th {:class (str disabled) :key title}
+                                 title])]]]
+        tbody [:tbody
+               [:<> (for [i (range (count (READ vpath)))]
+                      [:tr {:key i}
+                       [:<> (for [c columns]
+                              (make-column i c universals))]])]
+               (when sum-field [sum-field-component sum-field universals])]]
+    
     [:div.multi-table
-     (into [:table.table.is-striped.is-fullwidth.is-hoverable
-            headers
-            tbody])
-     (when-not disabled
-       [:div.control-buttons add-button delete-button])]))
+     [:table.table.is-striped.is-fullwidth.is-hoverable
```

Comment:
The classes on this table, besides `.table` which is common and descriptive, should be provided in a param. It's ok if the default is `["is-fulwidth" "is-hoverable"]`, though.  But make it customizable, not hard-coded like this